### ISSUE TITLE
clarify teamcity docs

### ISSUE THREAD

**Issue by trondhindenes** [marcells/node-build-monitor#112]
According to the docs, the teamcity integration needs a "build configuration id". Since this is something attached to each project it doesn't make sense - I would assume that node-build-monitor is able to monitor all builds on a teamcity server and not just a single one? I've tried setting it to "*" but that didn't do anything. It would be great if configuration items were clarified (both in terms of what they mean, and whether or not they're optional)

**Comment by trondhindenes** [comment]
From what I can see in the code, it looks like it's requesting a certain "id"
`/httpAuth/app/rest/buildTypes/id:' + self.configuration.buildConfigurationId +` but that kinda doesn't make sense, as the default should be to list all build configs? Teamcity's rest api is flexible enough to support some filtering, but not when the `id` slug is a hardcoded part of the request url.

**Comment by marcells** [comment]
We have to keep the option to specify a concrete build configuration. It could be that someone only wants to get one of hundreds configured builds on the server.

What we could do is allowing to omit the build configuration id config parameter. In that case we return all builds. So we are compatible to existing setups of the build monitor.

I don't have a Team City instance running. Do you want to make a PR with the change?

**Comment by trondhindenes** [comment]
I guess I'm a bit surprised that given other users of this project (from other issues I can see that there's at least a few teamcity shops) that this hasn't come up before. I'm simply not able to understand how this could work at all without a configuration block per project (We have about 200 of those) :-|. I'll take a closer look at this.

**Comment by marcells** [comment]
You are right. It's up to the implementation of the plugin how it can be configured. If there wasn't a need before, so it could be that it was not implemented yet for TeamCity.

The code of the TeamCity service is quite straight forward and has just 160 LOC. I can try to implement it, too. But this could take some time, cause I need to setup and configure a TeamCity instance.
